back/lib/tasks/single_use/20240530_migrate_visitor_timeline_widgets_craftjs_state.rake

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+namespace :single_use do
+  task migrate_visitor_timeline_widgets_craftjs_state: :environment do
+    # region HELPER METHODS
+    def migrate_layout(layout)
+      state = ContentBuilder::Craftjs::State.new(layout.craftjs_json)
+      widget_nodes = state.nodes_by_resolved_name('VisitorsWidget')
+
+      updated = widget_nodes.any? do |_node_id, node|
+        next false unless node['props'].key?('projectId')
+
+        node['props'].delete('projectId')
+        true
+      end
+
+      if updated
+        layout.save!
+
+        if layout.content_buildable_type == 'ReportBuilder::Report'
```

Comment:
These lines seem to be repeated a couple of times, would it make sense for the `Layout` or perhaps the `ContentBuilder::Craftjs::State` to have a `republish` method that does this?

Reply:
I'd like to keep `ContentBuilder::Craftjs::State` agnostic this kind of things, but it could be part of ` ReportBuilder::ReportPublisher`.